### DISCUSSION TITLE
Nesting a book into a website

### DISCUSSION THREAD

**Discussion by rahuldave** [quarto-dev/quarto-cli#3505] [Q&A]
I'd like to nest a book into a website. This lands up having 2 `_quarto.yml` files, and the one in the book's folder is not respected.

This is not surprising, as the one in the top level folder is being parsed.

I thought i might be able to add the contents of the nested `_quarto.yml` file to the top level one, but the project tag seems to be a singleton one.

Is there a way of achieving what i want to do?

**Comment by mcanouil** [comment]
I am not entirely sure to understand the desired result, but it seems the profile feature is what you need, especially the profile configuration.
See <https://quarto.org/docs/projects/profiles.html>.

Would you be able to share a small setup?

**Reply by AgedLace** [reply]
It is my understanding that @rahuldave wants to create a project type of "website", and then within that project also include the project type of "book".  From what I read in the profiles page, it doesn't address what @rahuldave is wanting to do.

**Reply by mcanouil** [reply]
Profiles help to switch between project level configuration file, e.g., one for a book and one for a website.
The downside is that to render everything, it requires the following step:
- switch to profile book
- render book
- switch to profile website
- render website

(Or website before book)

**Reply by rahuldave** [reply]
I did look into profiles, and in the interim settled upon a workflow that mirrors what you said but without using profiles. I generate two separate projects and splice the generated files from the book in. But @AgedLace is right...what i want to do is to generate everything from one `_quarto.yml`.

I'm trying to come up with a system that allows me to combine one-off projects that I undertake (and blog about) into "learning paths" or microbooks on the same site. For example, some stuff i have created on mlops, or testing, or just polymorphism. The content stays as a post in the blog, but now needs to be surrounded by different navigation for a microbook.

**Reply by mcanouil** [reply]
Then what you want to do is not possible using only Quarto I am afraid.

**Reply by rahuldave** [reply]
:-( Yes it does not seem like it for now! Iy seems like having a mode in which quarto processes can be on a stack may be a way to go...

**Comment by AgedLace** [comment]
While it may not currently be possible to actually incorporate a book[s] under a website / blog, I'm working on a process that *might* be helpful for you too.

By hosting a user / organization page on github [e.g. USER.github.io], any other project's repository can be accessed by appending it like so ... [USER.github.io/project].  

What I'm working on is creating my github user page as my quarto blog, then I will create a separate book [or other project] repository, and then create links in the navigation of both like that in the above example.  

The benefit of this is that multiple books / projects can be "linked' to the blog in this manner.  

I'm just now working on implementing it - not sure if it might be a potential interim solution for you as well.

**Reply by rahuldave** [reply]
I think it is! I implemented it at https://cudweed.univ.ai. There are two repos: the main site/blog one. But if you click on learning paths then it goes to another repo. I might need to manage this using git submodules.

One thing I couldnt figure out is how to get the title in the "book/learning_path" sub-page to point to the parent site...

**Reply by AgedLace** [reply]
So glad this idea was useful to you!  

As for the title link in the sub-page pointing to the parent ... would using an absolute link address work?

**Reply by AgedLace** [reply]
Forgot to add that I'm not up to speed on git submodules, but once I get the basics down with the blog and first book project, I'll delve into submodules more.  Thanks for the feedback!

**Reply by rahuldave** [reply]
Indeed you are right about absolute links. One needs to add the following  to the navbar:

```yaml
navbar:
    logo-href: https://cudweed.univ.ai
```

or wherever u want to go to (the root url of the parent site)

**Reply by AgedLace** [reply]
Thank you!! Keeping this for when I get to that part of my setup.

**Reply by rahuldave** [reply]
I also had to set `title:false` in the navbar or it insisted on putting the book title (naturally) rather than my use of logo:

```yaml
title: Hello
navbar:
    logo: https://cudweed.univ.ai/assets/bluelogo.svg
    logo-href: https://cudweed.univ.ai
    logo-alt: Univ.Ai
    title: false
    left:
      - href: https://cudweed.univ.ai/learningpaths.html
        text: Learning Paths
```

This is in the `_quarto.yml` for the subservient-to-main-site book

**Reply by AgedLace** [reply]
Thank you! Appreciate these tidbits!

**Reply by rahuldave** [reply]
There is something strange if i dont use absolute uris. Quarto insists on putting `./` before the uris. So i used absolute ones..

**Comment by michaelheilmann** [comment]
I do think the intended use of Quarto is to have separate projects for each publication and not nesting multiple publications. I also do think that nesting such different publications may cause troubles for the apparently much neglected multi-target publishing on many levels currently. Hence, the approach of @AgedLace seems to me to be be the correct one. I am following the separate projects approach as well to handle two books and an API documentation and a landing page glueing them together.

But this discussion is valuable as the current and the intended use of Quarto needs to be stated more clearly in the documentation in a timely fashion as the number of references to Quarto in which an abuse of Quarto - at least this is what I suspect it is - is propagated as a normal solution.

**Comment by cderv** [comment]
FWIW this website https://rstudio.github.io/r-manuals/ is a Quarto website, that is offering to read the R Manuals as Quarto books. 

The source code is here: https://github.com/rstudio/r-manuals

The "trick" was to generate the books as their own projects, with their own output folder, and then create a website that would have the rendered book folders as resources. 

This is quite meta and organize using scripts and templates. 

Drawback is that the whole Quarto website is not knowledgeable of the Books, meanings cross ref between books are non existent, and search is only available per-book, and each book is its own website. 

Just sharing in case in can give example to some with same needs

**Reply by michaelheilmann** [reply]
This is more or less what I am doing and this is - I suspect - what @AgedLace is doing.
Thanks for sharing (and confirming the validity of the approach and also its weaknesses).

**Comment by neeldhara** [comment]
I had the exact same question. My use case is that I use Quarto for [my website](www.neeldhara.com), which deploys via Netlify.

I am currently putting up lecture notes on page called [materials](www.neeldhara.com/materials), for example [here](https://www.neeldhara.com/materials/dsanotes/) is a specific set of notes (work in progress!). 

BTW, this seems to achieve some of @rahuldave's original use case, namely that "[...] content stays as a post in the blog, but now needs to be surrounded by different navigation for a microbook" --- this can be done with the sidebar directive in the main `_quarto.yml` file; and works fine for my lecture notes as well. What I feel I'll start missing soon is cross-referencing across chapters and PDF exports that respect that, since I'd eventually like for each of these lecture notes to evolve into books on their own right. 

So, it would be great to have nested book projects, but that is apparently not feasible, so I am wondering about the best workaround.

I could set this up as separate Quarto book projects, but ideally I'd like each book project to deploy to a dedicated subdomain (e.g. dsanotes.neeldhara.com or something like that). Netlify seems to support subdomains only via branch deploys: I am not sure I understand this correctly, but it seems like I'd have to have each set of lecture notes live in a parallel branch of my main git repo. This feels like it will be unwieldy --- or am I parsing this wrong?

Or I could do what's being done with the R Manuals site, but I can't seem to figure out how it is setup from their repo --- is there anywhere that the meta scripting and templating is documented?

**Reply by cderv** [reply]
> Or I could do what's being done with the R Manuals site, but I can't seem to figure out how it is setup from their repo --- is there anywhere that the meta scripting and templating is documented?

No it is not documented somewhere. The logic lives in the Rscript there. Basically, 

* We have a book project for each manuals. We render all those projects in there own output folder
* We move those output dirs into the website project, before rendering it. This means that website projects only has sub directories with complete HTML. 
* We use some templating mechanism for `_quarto.yml` and other files so that each book and the website share the same layout, theme and navbar. 

This is indeed meta but work ok. Just I believe the search is per book only. You can't search from website to a book in a subfolder. And also each project is independent, meaning you can't cross ref from one book to another. But we don't need that for R-Manuals. 

Hope it helps understand

**Reply by neeldhara** [reply]
Ah okay - got it - I think I can try to do something similar. I don't need references across books or searches to spawn multiple books either.

Thanks so much for your detailed response! I should be able to figure out the rest of the setup from the repo and will let you know if I have any specific questions.
